Tasks/DotNetCoreCLIV2/restorecommand.ts

```diff
@@ -179,6 +181,8 @@ function dotNetRestoreAsync(dotnetPath: string, projectFile: string, packagesDir
         dotnet.arg(verbosity);
     }
 
+    dotnet.line(args);
```

Comment:
Is it really necessary ? Because in the others commands (BasicCommands) which handle the `arguments` parameter it doesn't have any check.
https://github.com/microsoft/azure-pipelines-tasks/blob/b45ddada8afd4b720a9029c1c4489456d2c7c281/Tasks/DotNetCoreCLIV2/dotnetcore.ts#L104

If you really want it, for the implementation of the checks I think to do it like this.

```ts
   if (packagesDirectory) {
        dotnet.arg('--packages');
        dotnet.arg(packagesDirectory);

        if (args.includes('--packages') || args.includes('-p')) {
            tl.warning(tl.loc('DuplicatedArgument', '--packages'));
        }
    }

    dotnet.arg('--configfile');
    dotnet.arg(configFile);

    if (args.includes('--configfile')) {
        tl.warning(tl.loc('DuplicatedArgument', '--configfile'));
    }

    if (noCache) {
        dotnet.arg('--no-cache');

        if (args.includes('--no-cache')) {
            tl.warning(tl.loc('DuplicatedArgument', '--no-cache'));
        }
    }

    if (verbosity && verbosity !== '-') {
        dotnet.arg('--verbosity');
        dotnet.arg(verbosity);

        if (args.includes('--verbosity')) {
            tl.warning(tl.loc('DuplicatedArgument', '--verbosity'));
        }
    }
```

I doesn't have a way to query already added arguments, so I can't make the check later in a helper method.
Also I will need to add a new ressource string, how to handle this for all the language ?

Reply:
BasicCommands does not insert additional parameters based on the task settings, but restore does. This just makes it easier for users to spot the problem if the command starts failing.

You could use a helper like 

```ts
function warnIfArgsContains(args, ...values) {
  const argsArray = args.split(' ');
  for (var v of values) {
    if (argsArray.some(x => x == v)) {
      tl.warning(tl.loc('DuplicatedArgument', v));
    }
  }
}
```
Regarding resource strings, you need only add a resource string to the messages array in task.json.  When you build the task, the new resource will be added automatically to the US resource strings, and a localization team will later take care of the other languages.

Reply:
Thanks.
I made the requested changes.

Reply:
Thanks, looks great.